jv.c

```diff
@@ -728,15 +750,10 @@ jv jv_string_slice(jv j, int start, int end) {
   int c;
   jv res;
 
-  if (start < 0) start = len + start;
-  if (end < 0) end = len + end;
-
-  if (start < 0) start = 0;
-  if (start > len) start = len;
-  if (end > len) end = len;
-  if (end < start) end = start;
-  if (start < 0 || start > end || end > len)
+  if (!jvp_clamp_slice_params(len, &start, &end)) {
+    jv_free(j);
     return jv_invalid_with_msg(jv_string("Invalid string slice indices"));
```

Comment:
I noticed this while I was in here -- I believe returning here without calling `jv_free(j)` first is a leak.